developer/contributing/pull-requests.rst

```diff
@@ -32,67 +44,102 @@ If you are a member of the `Sulu organization`_ you should also:
+* If you answer yes to "BC breaks?", the pull request must contain updates to
+  the ``UPGRADE.md`` file.
+* If you answer yes to "Deprecations?", the pull request must contain updates to
+  the ``UPGRADE.md`` file.
+* If the "license" is not MIT, just don't submit the pull request as it won't
+  be accepted anyway.
 
 .. note::
 
   When an issue number is prefixed with ``fixes`` it tells GitHub to
-  automatically close the referenced ticket when the PR is merged
+  automatically close the referenced ticket when the pull request is merged
+
+What's in this PR?
+``````````````````
+
+Give as much details as possible about your changes (don't hesitate to give code
+examples to illustrate your points).
+The pull request description helps the code review and it serves as a reference
+when the code is merged.
 
-You can copy and paste the following template:
+Why?
+````
 
-.. code-block:: none
+If your pull request is about adding a new feature or modifying an existing one,
+explain the rationale for the changes. Why you added this feature, which problem
+does the PR fix?
 
-    [short description]:
+Example usage
+`````````````
 
-    __tasks:__
+If you add a new feature or make breaking changes, please give us a example
+usage (code examples, screenshots, etc.), so we understand what you try to
+solve.
 
-    - [ ] test coverage
-    - [ ] submit changes to the documentation
-    - [ ] ... <add your own tasks>
+Remove this section if not needed.
 
+BC Breaks/Deprecations
+``````````````````````
 
-    __informations:__
+If you have done BC breaks or deprecations, please describe them shortly here
+and add them to the ``UPGRADE.md`` file as well.
 
-    | q                | a
-    | ---------------- | ---
-    | Fixed tickets    |
-    | Related PRs      |
-    | BC breaks        |
-    | Documentation PR |
+Remove this section if not needed.
 
-Example
--------
+To Do
+`````
 
-Create user command:
+If some of the requirements in the checklist are not met, use the "To Do"
+section and add the relevant items:
 
-.. code-block:: none
+.. code-block:: text
 
-    __tasks:__
+    - [ ] Submit changes to the documentation
+    - [ ] Document the BC breaks
 
-    - [ ] test coverage
-    - [ ] submit changes to the documentation
-    - [ ] remove request defaults
-    - [ ] fix #1
+If the code is not finished yet because you don't have time to finish it or
+because you want early feedback on your work, add an item to to-do list:
 
+.. code-block:: text
 
-    __informations:__
+    - [ ] Finish the code
+    - [ ] Add tests as they have not been updated yet
+    - [ ] Gather feedback for my changes
 
-    | q                | a
-    | ---------------- | ---
-    | Fixed tickets    | fixes #1 , fixes #2
-    | Related PRs      | #3, #4
-    | BC breaks        | none
-    | Documentation PR | https://github.com/sulu-cmf/sulu-docs/pull/14
+Remove the "To Do" section if your pull request is complete and has no tasks.
```

Comment:
When thinking of it, I would not remove the Todo section, all the items should just be ticked before merging. The list might also contain information valuable for the reviewer (e.g. which parts of the PR has to be looked very closely at).